File: 프로그래머스/0/120875. 평행/평행.java
import java.util.*;

class Solution {
    double getSlope(int[] a, int[] b){
        int minusX = Math.abs(a[0] - b[0]);
        int minusY = Math.abs(a[1] - b[1]);
        return (double) minusY / minusX;
    }
    
    public int solution(int[][] dots) {
        if(getSlope(dots[0], dots[1]) == getSlope(dots[2], dots[3])){ return 1;}
        if(getSlope(dots[0], dots[2]) == getSlope(dots[1], dots[3])){ return 1;}
        if(getSlope(dots[0], dots[3]) == getSlope(dots[2], dots[1])){ return 1;}
           
        return 0;
    }
}

File: 프로그래머스/0/120875. 평행/평행.py
def solution(dots):
    def get_slope(dot1, dot2):
        x1, y1 = dot1[0], dot1[1]
        x2, y2 = dot2[0], dot2[1]
        return (y2 - y1) / (x2 - x1)

    cases = [
        [0, 1, 2, 3],
        [0, 2, 1, 3],
        [0, 3, 1, 2],
    ]

    for case in cases:
        a = dots[case[0]]
        b = dots[case[1]]
        c = dots[case[2]]
        d = dots[case[3]]
        if get_slope(a, b) == get_slope(c, d):
            return 1
    return 0